File: app/src/main/java/com/pluu/webtoon/adapter/viewholder/DaumNoticeImageViewHolder.java
package com.pluu.webtoon.adapter.viewholder;

import android.content.Context;
import android.graphics.Bitmap;
import android.view.View;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.animation.GlideAnimation;
import com.bumptech.glide.request.target.SimpleTarget;
import com.pluu.webtoon.R;
import com.pluu.webtoon.item.ChatView;
import com.pluu.webtoon.ui.view.AspectRatioImageView;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * 공지 이미지용 Item ViewHolder
 * Created by PLUUSYSTEM-SURFACE on 2016-05-21.
 */
public class DaumNoticeImageViewHolder extends BaseChattingViewHolder {
    @BindView(R.id.noticeImageView)
    AspectRatioImageView noticeImageView;

    public DaumNoticeImageViewHolder(View v) {
        super(v);
        ButterKnife.bind(this, v);
    }

    @Override
    public void bind(Context context, final ChatView item) {
        if (item.getHRatio() != 0) {
            noticeImageView.sethRatio(item.getHRatio());
            Glide.with(context)
                    .load(item.getImgUrl())
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
                    .into(noticeImageView);
        } else {
            Glide.with(context)
                    .load(item.getImgUrl())
                    .asBitmap()
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
                    .into(new SimpleTarget<Bitmap>() {
                        @Override
                        public void onResourceReady(Bitmap resource, GlideAnimation<? super Bitmap> glideAnimation) {
                            float hRatio = (float) resource.getHeight() / resource.getWidth();
                            item.setHRatio(hRatio);
                            noticeImageView.sethRatio(hRatio);
                            noticeImageView.setImageBitmap(resource);
                        }
                    });
        }
    }
}

File: app/src/main/java/com/pluu/webtoon/adapter/viewholder/DaumNoticeImageViewHolder.kt
package com.pluu.webtoon.adapter.viewholder

import android.content.Context
import android.graphics.Bitmap
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.SimpleTarget
import com.bumptech.glide.request.transition.Transition
import com.pluu.kotlin.inflate
import com.pluu.webtoon.R
import com.pluu.webtoon.item.ChatView
import kotlinx.android.synthetic.main.view_chatting_notice_image_layout.view.*

/**
 * 공지 이미지용 Item ViewHolder
 * Created by  pluu on 2017-05-02.
 */
class DaumNoticeImageViewHolder(v: View) : BaseChattingViewHolder(v) {

    override fun bind(context: Context, item: ChatView?) {
        if (item == null) return

        val option = RequestOptions().diskCacheStrategy(DiskCacheStrategy.NONE)
        if (item.hRatio != 0f) {
            itemView.noticeImageView.sethRatio(item.hRatio)
            Glide.with(context)
                .load(item.imgUrl)
                .apply(option)
                .into(itemView.noticeImageView)
        } else {
            Glide.with(context)
                .asBitmap()
                .load(item.imgUrl)
                .apply(option)
                .into(object : SimpleTarget<Bitmap>() {
                    override fun onResourceReady(
                        resource: Bitmap,
                        transition: Transition<in Bitmap>?
                    ) {
                        item.hRatio = resource.height.toFloat() / resource.width
                        itemView.noticeImageView.sethRatio(item.hRatio)
                        itemView.noticeImageView.setImageBitmap(resource)
                    }
                })
        }
    }

    companion object {
        fun newInstance(parent: ViewGroup) = DaumNoticeImageViewHolder(
            parent.inflate(R.layout.view_chatting_notice_image_layout, false)
        )
    }
}
